internal/cmd/argus/instance/create/create.go

```diff
@@ -0,0 +1,191 @@
+					return err
+				}
+			}
+
+			// Call API
+			req, err := buildRequest(ctx, model, apiClient)
+			if err != nil {
+				var dsaInvalidPlanError *cliErr.DSAInvalidPlanError
+				if !errors.As(err, &dsaInvalidPlanError) {
+					return fmt.Errorf("build Argus instance creation request: %w", err)
+				}
+				return err
+			}
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create Argus instance: %w", err)
+			}
+			instanceId := *resp.InstanceId
+
+			// Wait for async operation, if async mode not enabled
+			if !model.Async {
+				s := spinner.New(cmd)
+				s.Start("Creating instance")
+				_, err = wait.CreateInstanceWaitHandler(ctx, apiClient, instanceId, model.ProjectId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for Argus instance creation: %w", err)
+				}
+				s.Stop()
+			}
+
+			operationState := "Created"
+			if model.Async {
+				operationState = "Triggered creation of"
+			}
+			cmd.Printf("%s instance for project %q. Instance ID: %s\n", operationState, projectLabel, instanceId)
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().StringP(instanceNameFlag, "n", "", "Instance name")
+	cmd.Flags().Var(flags.UUIDFlag(), planIdFlag, "Plan ID")
+	cmd.Flags().String(planNameFlag, "", "Plan name")
+
+	err := flags.MarkFlagsRequired(cmd, instanceNameFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(cmd *cobra.Command) (*inputModel, error) {
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	planId := flags.FlagToStringPointer(cmd, planIdFlag)
+	planName := flags.FlagToStringValue(cmd, planNameFlag)
+
+	if planId == nil && (planName == "") {
+		return nil, &cliErr.DSAInputPlanError{
+			Cmd: cmd,
+		}
+	}
+	if planId != nil && (planName != "") {
+		return nil, &cliErr.DSAInputPlanError{
+			Cmd: cmd,
+		}
+	}
+
+	return &inputModel{
+		GlobalFlagModel: globalFlags,
+		InstanceName:    flags.FlagToStringPointer(cmd, instanceNameFlag),
+		PlanId:          planId,
+		PlanName:        planName,
+	}, nil
+}
+
+type argusClient interface {
+	CreateInstance(ctx context.Context, projectId string) argus.ApiCreateInstanceRequest
+	ListPlansExecute(ctx context.Context, projectId string) (*argus.PlansResponse, error)
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient argusClient) (argus.ApiCreateInstanceRequest, error) {
+	req := apiClient.CreateInstance(ctx, model.ProjectId)
+
+	var planId *string
+	var err error
+
+	plans, err := apiClient.ListPlansExecute(ctx, model.ProjectId)
+	if err != nil {
+		return req, fmt.Errorf("get Argus plans: %w", err)
+	}
+
+	if model.PlanId == nil {
+		planId, err = argusUtils.LoadPlanId(model.PlanName, plans)
+		if err != nil {
+			var dsaInvalidPlanError *cliErr.DSAInvalidPlanError
+			if !errors.As(err, &dsaInvalidPlanError) {
+				return req, fmt.Errorf("load plan ID: %w", err)
+			}
```

Comment:
Same here, we should create a `ArgusInvalidPlanError` for when the plan ID cannot be found for the given plan name